src/api/BaseApi.ts

```diff
@@ -0,0 +1,84 @@
+import { HTTPTransport } from "../services/HTTPTransport";
+import env from '../utils/env';
+
+interface IBaseApi {
+    baseUrl?: string,
+    path?: `/${string}`
+    headers?: Record<string, string>
+};
+
+
+const headers = {
+    'Content-type': 'application/json; charset=UTF-8',
+};
+
+const HTTP = new HTTPTransport();
+
+export class BaseAPI {
+
+    private _http: typeof HTTP;
+    private _baseUrl: string;
+    private _path: string;
+    private _headers: Record<string, string>;
+
+    constructor(config: IBaseApi = {}) {
+        this._http = HTTP;
```

Comment:
Тут тонкий момент, касающийся зависимостей. Сейчас класс BaseAPI не будет работать, если кто-то случайно изменит или удалит переменную HTTP. По хорошму, нужно прямо в конструкторе класса создавать инстанс HTTP сервиса
```suggestion
        this._http = new HTTPTransport();
```